### DISCUSSION TITLE
How to Create Tables in Snowflake using Terraform

### DISCUSSION THREAD

**Discussion by sfc-gh-dkotwani** [Snowflake-Labs/terraform-provider-snowflake#1003] [Q&A]
I am trying to create table in Snowflake using Terraform. Here is my code but the github action fails with an error: 
Run terraform fmt -check
  terraform fmt -check
  shell: /usr/bin/bash -e {0}
  env:
    TERRAFORM_CLI_PATH: /home/runner/work/_temp/e9948e8a-[2](https://github.com/sfc-gh-dkotwani/snow-repo/runs/6398672041?check_suite_focus=true#step:4:2)820-468b-98c4-6[3](https://github.com/sfc-gh-dkotwani/snow-repo/runs/6398672041?check_suite_focus=true#step:4:3)af95e3e002
/home/runner/work/_temp/e99[4](https://github.com/sfc-gh-dkotwani/snow-repo/runs/6398672041?check_suite_focus=true#step:4:4)8e8a-2820-468b-98c4-63af9[5](https://github.com/sfc-gh-dkotwani/snow-repo/runs/6398672041?check_suite_focus=true#step:4:5)e3e002/terraform-bin fmt -check
main.tf

Error: Terraform exited with code 3.
Error: Process completed with exit code 1.

Here is the code that I am using in main.tf file: 

terraform {
  required_providers {
    snowflake = {
      source  = "chanzuckerberg/snowflake"
      version = "0.25.17"
    }
  }

  backend "remote" {
    organization = "d-snow"

    workspaces {
      name = "gh-actions-demo"
    }
  }
}

provider "snowflake" {
}

resource "snowflake_database" "demo_Terraform_db" {
  name    = "DEMO_TERRAFORM_DB"
  comment = "Database for Snowflake Terraform demo"
}

resource "snowflake_schema" "demo_terraform_schema" {
  database = snowflake_database.demo_Terraform_db.name
  name     = "DEMO_TERRAFORM_SCHEMA"
  comment  = "Schema for Snowflake Terraform demo"
}

resource "snowflake_table" "table" {
  database            = snowflake_schema.DEMO_TERRAFORM_SCHEMA.DEMO_TERRAFORM_DB
  schema              = snowflake_schema.DEMO_TERRAFORM_SCHEMA.name
  name                = "table"
  comment             = "A table."
   change_tracking     = false

  column {
    name     = "id"
    type     = "int"
    nullable = true
 }

  column {
    name     = "identity"
    type     = "NUMBER(38,0)"
    nullable = true

    identity {
      start_num = 1
      step_num  = 3
    }
  }

  column {
    name     = "data"
    type     = "text"
    nullable = false
  }

  column {
    name = "DATE"
    type = "TIMESTAMP_NTZ(9)"
  }

  column {
    name    = "extra"
    type    = "VARIANT"
    comment = "extra data"
  }

  primary_key {
    name = "my_key"
    keys = ["data"]
  }
}

**Comment by Daniel-Giron** [comment]
This is really a problem with using the terraform fmt command incorrectly. The return code is not 0 because your HCL code is not formatted according to canonical form. If you want to format it, use the terraform fmt command with the -check option.
This issue can be closed.
